### ISSUE TITLE
Training finishes immediately without reporting "finished" (finetuning for a new dataset)

### ISSUE THREAD

**Issue by fperezgamonal** [facebookresearch/maskrcnn-benchmark#584]
## ❓ Questions and Help

Hi there,

I am trying to fine-tune a pre-trained model for one sequence of the [AICity challenge](https://www.aicitychallenge.org/) with only two classes from the [Coco dataset](http://cocodataset.org/#format-data) ('__background__' and '__car__'). I kept the 'category_id': 3 for all car annotations and defined the categories list with only one dictionary for the class car, following the detection-specific format on top of the fields included as default (info, licenses, etc.) :
```
"categories": [{"id": 3, "name": "car", "supercategory": "vehicle"}]
```` 

The main steps I followed to be able to fine-tune to my dataset, were:

- Follow all the steps in #521 (including trimming the pre-trained model' last layer and linking that to the training config file)
- Also changed the checkpoint.py function as in #273 

However, when I run the training script as:
```
python tools/train_net.py --config-file configs/caffe2/ours_e2e_faster_rcnn_R_50_FPN_1x_caffe2.yaml
```

the training script loads the correct config file and displays the complete list of training params (pre-defined + default ones) with the correct value, that is, those modified by the config remain the same (e.g.: MODEL.ROI_BOX_HEAD.NUM_CLASSES: 2, and WEIGHT pointing to the correct path of the trimmed weights) and then it starts training but prints the following and exits:
```
loading annotations into memory...
Done (t=0.01s)
creating index...
index created!
2019-03-22 11:39:58,913 maskrcnn_benchmark.trainer INFO: Start training
max_iter(len(data_loader)) = 90000
start_iter: 0
getting item 34
getting item 292
getting item 87
getting item 157
getting item 262
getting item 325
For (enumerate(data_loader, start_iter): 
images.image_sizes: [torch.Size([750, 1333])]
targets: (BoxList(num_boxes=2, image_width=1333, image_height=750, mode=xyxy),)
target: BoxList(num_boxes=2, image_width=1333, image_height=750, mode=xyxy)
getting item 270
getting item 203
getting item 298
```
By taking a look at the log.txt saved in the output directory I set, the prints "getting item NUM", do not appear so that makes me think it is related to the format of the annotations (conflicting with Coco format) but I could not find where this is printed from. **(\**)**

Another thing to (maybe) consider is that I trimmed the weights as follows (with [this script](https://gist.github.com/wangg12/aea194aa6ab6a4de088f14ee193fd968)):
```
!python trim_detectron_model.py \
--pretrained_path /root/.torch/models/_detectron_35857345_12_2017_baselines_e2e_faster_rcnn_R-50-FPN_1x.yaml.01_36_30.cUF7QR7I_output_train_coco_2014_train%3Acoco_2014_valminusminival_generalized_rcnn_model_final.pkl \
--save_path ../weights/trimmed_weights/trimmed_e2e_faster_rcnn_R_50_FPN_1x_OURconfig.pth \
--cfg ../configs/caffe2/e2e_faster_rcnn_R_50_FPN_1x_caffe2.yaml
```
where the pretrained weights for FasterRCNN R-50 FPN1x were downloaded from [here](https://dl.fbaipublicfiles.com/detectron/35857345/12_2017_baselines/e2e_faster_rcnn_R-50-FPN_1x.yaml.01_36_30.cUF7QR7I/output/train/coco_2014_train%3Acoco_2014_valminusminival/generalized_rcnn/model_final.pkl) and the configuration file is the one used in the demo which downloads the weights mentioned above.
Notice that when trimming, the config file did not specify NUM_CLASSES: 2, etc., I did that only in the one passed to training (I do not know if that is the right way or not).
To check this, I will try to only load from the pre-trained weights, the model itself and not the solver configuration, etc. as suggested in #372 **(*)**.

The config file provided for training is adapted from the one used in trimming and looks like this:
```
MODEL:
  META_ARCHITECTURE: "GeneralizedRCNN"
  WEIGHT: "/content/drive/ColabNotebooks/maskrcnn-benchmark/weights/trimmed_weights/trimmed_e2e_faster_rcnn_R_50_FPN_1x_ogconfig.pth"
  BACKBONE:
    CONV_BODY: "R-50-FPN"
  RESNETS:
    BACKBONE_OUT_CHANNELS: 256
  RPN:
    USE_FPN: True
    ANCHOR_STRIDE: (4, 8, 16, 32, 64)
    PRE_NMS_TOP_N_TRAIN: 2000
    PRE_NMS_TOP_N_TEST: 1000
    POST_NMS_TOP_N_TEST: 1000
    FPN_POST_NMS_TOP_N_TEST: 1000
  ROI_HEADS:
    USE_FPN: True
  ROI_BOX_HEAD:
    POOLER_RESOLUTION: 7
    POOLER_SCALES: (0.25, 0.125, 0.0625, 0.03125)
    POOLER_SAMPLING_RATIO: 2
    FEATURE_EXTRACTOR: "FPN2MLPFeatureExtractor"
    PREDICTOR: "FPNPredictor"
    NUM_CLASSES: 2
DATASETS:
  TRAIN: ("aicity_2019_train", )
  TEST: ("aicity_2019_val", )
DATALOADER:
  SIZE_DIVISIBILITY: 32
SOLVER:
  BASE_LR: 0.0001
  WEIGHT_DECAY: 0.0001
  STEPS: (70000,)
  IMS_PER_BATCH: 1
  MAX_ITER: 90000
TEST:
  IMS_PER_BATCH: 1
OUTPUT_DIR: "/content/drive/ColabNotebooks/maskrcnn-benchmark/results/AICITY_finetuning_FastRCNN/"
```

I attach the complete log below in case it may be helpful.
[log.txt](https://github.com/facebookresearch/maskrcnn-benchmark/files/2992081/log.txt)
PS: I am working in Google Colab with 1 GPU.

Any help would be appreciated,
Thanks in advance,
Ferran.

**(*) UPDATE:** I tried to trim the model with the same config file as the one provided for training but the result is the same print out (with different indexes because samples are pick randomly). I also noticed that only 8 "getting item..." statements are printed, which has to be related to the batch size being equal to 8, I guess. 

**(\**) UPDATE2:** I added a couple of print outs to display information about image_list, boxlist and starting and maximum iterations (length of the data_loader). I have also been able to correctly load and display my annotations on some images by using the CocoAPI in Python so it is unlikely related to the annotations format.

**Comment by chengyangfu** [comment]
Hi  @fperezgamonal, 
   I think the trimmed model is loaded properly according to the log and the problem could be the dataset format. 

   This is what I will do. Using the Pythong Debug to check what the images and annotations feed to the network. 
   If you are familiar with the Python debug, setting a stopping point at [here](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/engine/trainer.py#L63) to see the what the images and annotations.  Then run line by line to see when the program break.  For example, in forwarding passing or backpropagation.

**Comment by fperezgamonal** [comment]
Hello @chengyangfu ,

Sorry for the late reply but we were trying to make this code work for a Master project and we had to finish more work as well. I'll definitely try your suggestion if I use Mask R-CNN in the future to see where the fork to add a new dataset breaks.

I close this issue but feel free to re-open it if you think it is better for other users.

Thanks!
